cli/commands/wp.ts

```diff
@@ -1,81 +1,113 @@
-import { __ } from '@wordpress/i18n';
+import { SupportedPHPVersion, SupportedPHPVersions } from '@php-wasm/universal';
+import { __, sprintf } from '@wordpress/i18n';
 import { ArgumentsCamelCase } from 'yargs';
 import yargsParser from 'yargs-parser';
+import { z } from 'zod';
 import { getSiteByFolder } from 'cli/lib/appdata';
 import { connect, disconnect } from 'cli/lib/pm2-manager';
 import { runWpCliCommand } from 'cli/lib/run-wp-cli-command';
-import { validatePhpVersion } from 'cli/lib/utils';
 import { isServerRunning, sendWpCliCommand } from 'cli/lib/wordpress-server-manager';
 import { Logger, LoggerError } from 'cli/logger';
 import { GlobalOptions } from 'cli/types';
 
 const logger = new Logger< '' >();
 
-export async function runCommand( siteFolder: string, args: string[] ): Promise< void > {
+export async function runCommand(
+	siteFolder: string,
+	args: string[],
+	options: {
+		phpVersion?: string;
+	} = {}
+): Promise< void > {
 	const site = await getSiteByFolder( siteFolder );
 
-	// If there's already a running Playground instance for this site, pass the command to it…
+	// Determine the PHP version to use
+	const phpVersionSchema = z.enum( SupportedPHPVersions );
+	let phpVersion: SupportedPHPVersion;
+	const requestedPhpVersion = options.phpVersion ?? site.phpVersion;
+
 	try {
-		await connect();
+		phpVersion = phpVersionSchema.parse( requestedPhpVersion );
+	} catch ( error ) {
+		throw new LoggerError( sprintf( __( 'Unsupported PHP version: %s' ), requestedPhpVersion ) );
+	}
```

Comment:
```ts
	const phpVersion = validatePhpVersion( options.phpVersion ?? site.phpVersion );
```
I would move the `catch` clause to `validatePhpVersion` and keep using that helper function to encapsulate this boilerplate-y logic.